frameworks/MLPlan/__init__.py

```diff
@@ -0,0 +1,22 @@
+from amlb.benchmark import TaskConfig
+from amlb.data import Dataset
+from amlb.resources import config as rconfig
+from amlb.utils import call_script_in_same_dir
+
+
+def setup(*args, **kwargs):
+    call_script_in_same_dir(__file__, "setup.sh", rconfig().root_dir, *args, **kwargs)
+
+
+def run(dataset: Dataset, config: TaskConfig):
+    from frameworks.shared.caller import run_in_venv
+
+    data = dict(
+        train=dict(path=dataset.train.path),
+        test=dict(path=dataset.test.path),
+        target=dict(index=dataset.target.index),
+        predictors_type=['Numerical' if p.is_numerical() else 'Categorical' for p in dataset.predictors]
+    )
+
+    return run_in_venv(__file__, "exec.py",
+                       input_data=data, dataset=dataset, config=config)
```

Comment:
This should solve your reordering issue:
```suggestion
from amlb.benchmark import TaskConfig
from amlb.data import Dataset
from amlb.resources import config as rconfig
from amlb.datautils import reorder_dataset
from amlb.utils import call_script_in_same_dir


def setup(*args, **kwargs):
    call_script_in_same_dir(__file__, "setup.sh", rconfig().root_dir, *args, **kwargs)


def run(dataset: Dataset, config: TaskConfig):
    from frameworks.shared.caller import run_in_venv

    train_path = dataset.train.path
    test_path = dataset.test.path
    mode = config.framework_params.get('_mode')
    # Weka requires target as the last attribute
    if mode == 'weka' and dataset.target.index != len(dataset.predictors):
        train_path = reorder_dataset(dataset.train.path, target_src=dataset.target.index)
        test_path = reorder_dataset(dataset.test.path, target_src=dataset.target.index)

    data = dict(
        train=dict(path=train_path),
        test=dict(path=test_path),
        target=dict(index=dataset.target.index),
    )

    return run_in_venv(__file__, "exec.py",
                       input_data=data, dataset=dataset, config=config)
```